main.py

```diff
@@ -946,10 +966,13 @@ def _uni_from_dict(self, uni_dict):
                 + f"Could not instantiate interface {interface_id}"
             )
             raise ValueError(result) from ValueError
-
+        tag_convert = {1: "vlan"}
         tag_dict = uni_dict.get("tag", None)
         if tag_dict:
-            tag = TAG.from_dict(tag_dict)
+            tag_type = tag_dict.get("tag_type")
+            tag_type = tag_convert.get(tag_type, tag_type)
+            tag_value = tag_dict.get("value")
+            tag = TAG(tag_type, tag_value)
```

Comment:
This addition is to still accept TAG integer, `openapi` is accepting integer and string (`enum: [1, "vlan"]`). Since every API call goes through this method, this is enough to transform every integer to `"vlan"`.
Ccing @italovalcy

Reply:
@Alopalao very nice, and the data model is using great using (vlan that's easier to reason and maintain in the future). However, if the point of still allowing in the API `tag_type` 1 not to break existing scripts and clients, which is a very reasonable trade-off and compromise, then we probably also don't want to bump v3, otherwise it'll break anyway, which would partly defeat the purpose of compatibility. Keeping as v2 while allowing "vlan" and "vlan" being the new value in the data model arguably would require v3, but we can be pragmatic all things considered, keep as v2, and only mention in the changelog. It's up to you guys.

Reply:
The conclusion was to keep the version, `vlan` input as string is an addition to integer `tag_type`. The output and the saved form of `tag_type` is as string.